doc/website/getting-started/overview.md

```diff
@@ -0,0 +1,540 @@
+which in turn will wake up a potentially waiting thread. In this way it extends a condition variable to a collection
+of conditions. Upon waking up it can be determined which conditions became true and caused the wake up. In the case
+that the wake up event was the availability of new data, this data can now be collected at the subscriber.
+
+For more information on how to use the Waitset see [Waitset](../../../iceoryx_examples/waitset/README.md).
+
+## API
+
+Now, we show how the API can be used to establish a publish-subscribe communication in an iceoryx system. Many parts
+of the API follow a functional programming approach and allow the user to specify functions which handle the possible
+cases, e.g. what should happen when data is received.
+
+This is very flexible but requires using the monadic types ``cxx::expected`` and ``cxx::optional``, which we
+introduce in the following sections.
+
+We distinguish between the ``typed API`` and the ``untyped API``. In the Typed API the underlying data type is made
+apparent by typed pointers or references to some data type T (often a template parameter). this allows working with
+the data in an C++ idiomatic and type-safe way and should be preferred whenever possible.
+
+The Untyped API provides opaque (i.e. void) pointers to data, which is flexible and efficient but also requires that
+the user takes care to interpret received data correctly, i.e. as a type compatible to what was actually sent. This
+is required for interaction with other lower level APIs and should be used sparingly. For further information see the
+respective header files.
+
+There also is a plain [C API](../../../iceoryx_examples/icedelivery_on_c/README.md), which can be used if C++ is not
+an option.
+
+We now describe the how to use the API in iceoryx applications. We will omit namespaces in several places to keep
+the code concise. In most cases it can be assumed that we are using namespace ``iox::cxx``. We also will use ``auto``
+sparingly to clearly show which types are involved, but in many cases automatic type deduction is possible and can
+shorten the code.
+
+### Optional
+
+The type ``cxx::optional<T>`` is used to indicate that there may or may not be a value of a specific type ``T``
+available. This is essentially the maybe monad in functional programming. Assuming we have some optional (usually the
+result of some computation)
+```
+optional<int> result = someComputation();
+```
+we can check for its value using
+```
+if(result.has_value())
+{
+    auto value = result.value();
+    //do something with the value
+} else {
+    //handle the case that there is no value
+}
+```
+A shorthand to get the value is 
+```
+auto value = *result;
+```
+
+Note that getting the value if there is no value is undefined behavior, so it must be checked beforehand.
+
+We can achieve the same with the functional approach by providing a function for both cases.
+
+```
+result.and_then([](int& value) { /*do something with the value*/ })
```

Comment:
Could you format the code in-line with our clang-format?

Reply:
I do not like this, it stretches it out over too much lines.

Reply:
For me the stretching is not an issue. @dkroenke What do you think?

Reply:
Now I would write something like this always in one line, and clang format keeps it like this (since it is short). It is only when you start with multiple lines that it reformats. 
I encourage all coders also to think about economy of screen space.

We do not want code were people are playing code golf and cram up anything in one line, but this is not the case here.
We also do not want code were trivial aspects are stretched over multiple line when it is perfectly readable on one. 

This is only briefly mentioned somewhere in this talk of Walter E. Brown, but the talk is useful and entertaining enough to list it here:
https://www.youtube.com/watch?v=OAmWHmwlMwI
I am too lazy to look up were it is mentioned in the talk, but the reasoning is as above.

He is a cool guy (see also https://www.youtube.com/watch?v=OAmWHmwlMwI , he worked with many influential people) and while I do not agree with everything in this talk, on the aspect of screen economy I do.